### DISCUSSION TITLE
[Feature Request] New component : Tree view

### DISCUSSION THREAD

**Discussion by AntoineArt** [heroui-inc/heroui#1262] [Feature requests]
### Is your feature request related to a problem? Please describe.

For some projects we need to have an arborescent way of showing informations. But the available solutions are ugly and not practical.

### Describe the solution you'd like

I would need a tree view in next UI with similar functionalities as MUI's treeview, but pretty and easy to use.

An important feature is to be able to trigger a function when an item is clicked.

### Describe alternatives you've considered

Here is MUI's treeview : https://mui.com/material-ui/react-tree-view/
Here is an example (the tree view is the first column) : 
![image](https://github.com/nextui-org/nextui/assets/26147209/4b9e63c1-b455-432b-bf3d-15e8d55ea620)

**Comment by Jerrykey1369** [comment]
Just thinking if you change the view by making everything folding a different direction to appeal to someone else's ability to see things backwards. Shooting at the wall.

**Reply by AntoineArt** [reply]
Hmmm can you elaborate ? I don't understand what you are saying 😅

**Comment by jguddas** [comment]
https://spectrum.adobe.com/page/tree-view

**Comment by nikita-fxbytes** [comment]
How to implement this type of view in NextUI?
![Screenshot from 2024-02-02 13-01-17](https://github.com/nextui-org/nextui/assets/99159940/d3ee937a-16a0-4b65-b03b-d31b48a8a9a1)

**Comment by haidar47x** [comment]
I'm working on a custom tree based on React ARIA but having a tree in HeroUI would be nice.
